### PR TITLE
Validate HostConfig.extraHosts

### LINKED ISSUE #699: extraHosts(): check parameters to shield from docker daemon panic
With docker version `1.12.6, build 78d1802` and docker-client 8.1.3-SNAPSHOT from ~1 week ago, I ran into [docker issue #20997](https://github.com/docker/docker/issues/20997) when supplying wrong parameters to the `hostConfig`'s `extraHosts` parameter.

E.g., when wrongly passing in `extraHosts("host", "alias")` instead of `...("host:alias")`, as I mistakenly did, I'd prefer to get an IllegalArgumentException because the host alias is not properly formatted.

Although it is a docker daemon bug (producing an `out of bounds` error instead of returning the proper cause (illegal argument), it would save devs some time when the library could check the parameters first.

### PR DESCRIPTION
**Pull request by davidxia**
Extra hosts args must contain ":".

fixes #699

### CODE REVIEW

**Comment by davidxia** [comment]
cc @grexe

**Comment by davidxia** [comment]
@mattnworb Did you get a chance to take a look?

File: src/main/java/com/spotify/docker/client/messages/HostConfig.java [hunk 1]
```diff
@@ -627,7 +628,22 @@ public Builder hostPidMode() {
     
     public abstract Builder storageOpt(Map<String, String> tmpfs);
 
-    public abstract HostConfig build();
+    abstract HostConfig autoBuild();
```
Comment (mattnworb) [inline src/main/java/com/spotify/docker/client/messages/HostConfig.java]: should leave a comment about why this exists, it isn't very obvious

File: src/main/java/com/spotify/docker/client/messages/HostConfig.java [hunk 2]
```diff
@@ -627,7 +628,22 @@ public Builder hostPidMode() {
...
+      validateExtraHosts(hostConfig.extraHosts());
+      return hostConfig;
+    }
+  }
+
+  private static void validateExtraHosts(final List<String> extraHosts) {
+    if (extraHosts != null) {
+      for (final String extraHost : extraHosts) {
+        checkArgument(extraHost.contains(":"),
+            String.format("extra host arg '%s' must contain a ':'", extraHost));
```
Comment (mattnworb) [inline src/main/java/com/spotify/docker/client/messages/HostConfig.java]: no need to call String.format yourself, there is an overload of checkArgument that does:

```
public static void checkArgument(boolean expression,
                 @Nullable
                 String errorMessageTemplate,
                 @Nullable
                 Object... errorMessageArgs)
```

Comment (davidxia) [inline src/main/java/com/spotify/docker/client/messages/HostConfig.java]: thanks. Didn't notice this.

File: src/main/java/com/spotify/docker/client/messages/HostConfig.java (hunk 1 above)
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/messages/HostConfig.java]: Done.

**Comment by davidxia** [comment]
Addressed both comments. How's it look now?

**Pull request merged**